nodejs/week3/homework-template/src/warmup/index.js

```diff
@@ -0,0 +1,101 @@
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+app.set('json spaces', 2); // prettify json output
+
+app.get("/calculator/:method", (req, res) => {
+  if (req.params.method === "add") {
+    let result = addNumbers(req);
+    res.send("Sum is:" + result);
+  } else if (req.params.method === "subtract") {
+    const diff = subtract(req);
+    res.send("Result is:" + diff);
+  } else if (req.params.method === "multiply") {
+    const result = multiply(req);
+    res.send("Result is:" + result);
+  } else if (req.params.method === "divide") {
+    const result = divide(req);
+    res.send("Result is:" + result);
+  }
+});
+
+//Calculator using req.body
+app.post("/calculator", (req, res) => {
+  if (req.body.method === "add") {
+    let result = addNumbers(req);
+    res.send("Sum is:" + result);
+  } else if (req.body.method === "subtract") {
+    const diff = subtract(req);
+    res.send("Result is:" + diff);
+  } else if (req.body.method === "multiply") {
+    const result = multiply(req);
+    res.send("Result is:" + result);
+  } else if (req.body.method === "divide") {
+    const result = divide(req);
+    res.send("Result is:" + result);
+  }
+});
+
+function divide(req) {
+  let result = 1;
+  for (var param in req.query) {
+    result = Number(req.query[param]) / result;
+  }
+  return result;
+}
+
+function multiply(req) {
+  let result = 1;
+  for (var param in req.query) {
+    if (isNaN(req.query[param])) {
+      const arrayOfParams = req.query[param];
+      result =
```

Comment:
Okay, so i know this looks really fancy, **but** when i as a developer comes across such a line i need to use a lot of mental power to figure out what it does. That is usually a bad sign. especially when what you are doing is rather simle

You can compare it to:
```
const multiplicationResult = 0;
for (var param in req.query) {
  if (isNaN(req.query[param])) {
    multiplicationResult  *= req.query[param];
  }
}
```
Its the simpler way, not as fancy, but the readability is higher, because it is simpler. So when you code KISS (Keep It Stupid Simple)